WordPressUI/Extensions/UIImageView+Gravatar.swift

```diff
@@ -131,23 +171,31 @@ extension UIImageView {
     ///
     /// - Returns: Gravatar's URL
     ///
-    private func gravatarUrl(for email: String, size: Int, rating: String) -> URL? {
-        let sanitizedEmail = email
+    private func gravatarUrl(for email: String, size: Int, rating: GravatarRatings) -> URL? {
+        let hash = gravatarHash(of: email)
+        let targetURL = String(format: "%@/%@?d=404&s=%d&r=%@", Defaults.baseURL, hash, size, rating.stringValue())
+        return URL(string: targetURL)
+    }
+
+    /// Returns the gravatar hash of an email
+    ///
+    /// - Parameter email: the email associated with the gravatar
+    /// - Returns: hashed email
+    ///
+    /// This really ought to be in a different place, like Gravatar.swift, but there's
+    /// lots of duplication around gravatars -nh
+    private func gravatarHash(of email: String) -> String {
+        return email
             .lowercased()
             .trimmingCharacters(in: .whitespaces)
-        let targetURL = String(format: "%@/%@?d=404&s=%d&r=%@", Defaults.baseURL, sanitizedEmail.md5(), size, rating)
-        return URL(string: targetURL)
+            .md5()
     }
 
     /// Returns the required gravatar size. If the current view's size is zero, falls back to the default size.
     ///
     private func gravatarDefaultSize() -> Int {
-        guard bounds.size.equalTo(.zero) == false else {
-            return Defaults.imageSize
-        }
-
-        let targetSize = max(bounds.width, bounds.height) * UIScreen.main.scale
-        return Int(targetSize)
```

Comment:
Nate, what's the reason behind this change? We were downloading an image to fit the size of the UIImageView. With this change we should see slower downloads, and higher bandwidth usage. Can we revert this one?

Reply:
For the notification to work, we need to download a size that works in multiple locations. We'd be downloading 80px * display res, which is 240px max. I figured it wasn't a problem. Without this change, the solution becomes more complicated…